### DISCUSSION TITLE
How to best import esprima-next into an ES6 module?

### DISCUSSION THREAD

**Discussion by nikhilasm** [node-projects/esprima-next#31] [Q&A]
Hi there, I'm new to this library and pretty new to JS modules in general, but I'm running into issues trying to load this library in my project. My project is an ESM, with the ```"type": "module"``` set in the ```package.json``` file and everything.

If I try to import the ```parseModule``` method using the destructuring export syntax, i.e. ```import { parseModule } from 'esprima-next';```, I get the following error:

SyntaxError: Named export 'parseModule' not found. The requested module 'esprima-next' is a CommonJS module, which may not support all module.exports as named exports.

I've tried all sorts of combinations of ```import``` and ```require```, but I can't seem to get it working. The only solution I've found is to add the ```"type": "module"``` line to the ```package.json``` file inside of my local copy of the esprima-next node module. Is there something I'm missing?

Thanks for your help!

**Comment by jogibear9988** [comment]
Are you using JS or Typescript? If Typescript, wich Version?
NodeJs Version?
Is it for frontend or backend?
How does your tsconfig look?

**Reply by nikhilasm** [reply]
Thanks for your reply. It might be a strange use-case, but I'm building a standard JS command-line application (so all backend, basically) using Node v16.13.2. Since I'm not using TypeScript, I don't have a tsconfig, but my Node package.json is very basic.

**Comment by jogibear9988** [comment]
don't know if node v16 supports "exports" in package.json
maybe you need to import from esm subdirectory
As we also deliver requireJs package.
Maybe I should remove this. This should be task of the bundler.
see https://github.com/node-projects/esprima-next/blob/bc2b9831f772fad96b09762fe82ba6432caadb73/package.json#L11

**Reply by nikhilasm** [reply]
I see. I tried to import as such ```import esprima from 'esprima-next/dist/esm/esprima.js';```, but still no luck. I now get an error saying that path is not defined by "exports" in the esprima-next/package.json file.
